lib/web/ui/cluster.go

```diff
@@ -17,36 +17,72 @@ limitations under the License.
 package ui
 
 import (
+	"fmt"
 	"sort"
 	"time"
 
+	"github.com/gravitational/teleport/lib/defaults"
 	"github.com/gravitational/teleport/lib/reversetunnel"
+	"github.com/gravitational/trace"
 )
 
 // Cluster describes a cluster
 type Cluster struct {
 	// Name is the cluster name
 	Name string `json:"name"`
 	// LastConnected is the cluster last connected time
-	LastConnected time.Time `json:"last_connected"`
+	LastConnected time.Time `json:"lastConnected"`
 	// Status is the cluster status
 	Status string `json:"status"`
+	// NodeCount is this cluster number of registered servers
+	NodeCount int `json:"nodeCount"`
+	// PublicURL is this cluster public URL (its first available proxy URL)
+	PublicURL string `json:"publicURL"`
 }
 
-// NewClusters creates a slice of Clusters
-func NewClusters(remoteClusters []reversetunnel.RemoteSite) []Cluster {
+// NewClusters creates a slice of Cluster's, containing data about each cluster.
+func NewClusters(remoteClusters []reversetunnel.RemoteSite) ([]Cluster, error) {
 	clusters := []Cluster{}
-	for _, item := range remoteClusters {
+	for _, rclsr := range remoteClusters {
+		clt, err := rclsr.CachingAccessPoint()
+		if err != nil {
+			return nil, trace.Wrap(err)
+		}
+
+		// public URL would be the first connected proxy public URL
+		proxies, err := clt.GetProxies()
+		if err != nil {
+			return nil, trace.Wrap(err)
+		}
+
+		publicURL := ""
+		if len(proxies) > 0 {
+			publicURL = proxies[0].GetPublicAddr()
+
+			// if public_address under proxy_service in config is not set, it is empty
+			// in this case we manually build the URL using proxy's hostname and default port
+			if publicURL == "" {
+				publicURL = fmt.Sprintf("%v:%v", proxies[0].GetHostname(), defaults.HTTPListenPort)
+			}
+		}
```

Comment:
Similar logic exists here: https://github.com/gravitational/teleport/blob/master/lib/web/apiserver.go#L2134-L2158

Can we consolidate this to a function and call it somewhere? That way we return an consistent answer always?

Reply:
@kimlisa Once you've added the above function, and you leave a comment in https://github.com/gravitational/teleport/pull/3485, we need it there as well.

Reply:
@russjones this PR is against `hornet` branch which are not going to be merged with the master any time soon.

@kimlisa could you refactor it to a separate function without worrying about #3485?